src/main/java/lotto/domain/WinningLotto.java

```diff
@@ -0,0 +1,36 @@
+package lotto.domain;
+
+public class WinningLotto {
+    private final int ADD_COUNT = 1;
+    private final int ZERO = 0;
+
+    private Lotto lotto;
+    private int bonusNumber;
+
+    public WinningLotto(Lotto lotto, String bonusNumber) {
+        this.lotto = lotto;
+        this.bonusNumber = Integer.parseInt(bonusNumber);
+    }
+
+    public int getMatchCount(Lotto lotto) {
+        int matchCount = 0;
+
+        for (Integer number : this.lotto.getNumbers()) {
+            matchCount += addMatchCount(lotto, number);
+        }
+
+        return matchCount;
+    }
```

Comment:
```suggestion
    public int getMatchCount(Lotto lotto) {
        return this.lotto.match(lotto);
    }
```
값을 꺼내와서 연산하기보다는 해당 객체에 메시지를 보내보는 것은 어떨까요
getter를 사용할 때는 코드스멜로 보는 것도 좋다고 생각합니다.
getter 메서드를 사용하게 되면 캡슐화가 깨질 가능성이 높아집니다.
getter 메서드를 사용함으로써 외부에 public으로 해당 변수가 존재한다는 사실을 공개하게 되는데요.
이렇게 내부 데이터에 초점을 두고 프로그래밍 할 경우에 내부의 변화가 외부로 전파될 경우가 발생하게 됩니다.

---

```java
    public int match(Lotto lotto) {
        return (int) numbers.stream()
                .filter(number -> lotto.contains(number))
                .count();
    }
```

Stream API를 활용하여 반복문을 내부라이브러리에 위임함으로써 연산을 드러내는 것은 어떨까요